jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamic/DynamicResourceFileActionProvider.kt

```diff
@@ -11,37 +11,41 @@ import com.intellij.openapi.util.Key
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.ui.EditorNotificationPanel
 import com.intellij.ui.EditorNotifications
+import software.aws.toolkits.jetbrains.AwsToolkit
 import software.aws.toolkits.resources.message
 
 class DynamicResourceFileActionProvider :
     EditorNotifications.Provider<DynamicResourceFileActionProvider.DynamicResourceVirtualFilePanel>() {
     override fun getKey(): Key<DynamicResourceVirtualFilePanel> = KEY
 
     override fun createNotificationPanel(file: VirtualFile, fileEditor: FileEditor, project: Project):
-        DynamicResourceVirtualFilePanel? = when (file) {
-        is CreateDynamicResourceVirtualFile ->
-            DynamicResourceVirtualFilePanel(
-                project,
-                file,
-                message("dynamic_resources.create_resource_instruction"),
-                "dynamic.resource.editor.submitResourceCreationRequest"
-            )
-        is ViewEditableDynamicResourceVirtualFile ->
-            when (file.isWritable) {
-                true -> DynamicResourceVirtualFilePanel(
+        DynamicResourceVirtualFilePanel? {
+        if (!AwsToolkit.isMoreResourcesMutationEnabled()) return null
+        return when (file) {
+            is CreateDynamicResourceVirtualFile ->
+                DynamicResourceVirtualFilePanel(
                     project,
                     file,
-                    message("dynamic_resources.update_resource_instruction"),
-                    "dynamic.resource.editor.submitResourceUpdateRequest"
+                    message("dynamic_resources.create_resource_instruction"),
+                    "dynamic.resource.editor.submitResourceCreationRequest"
                 )
-                false -> DynamicResourceVirtualFilePanel(
-                    project,
-                    file,
-                    message("dynamic_resources.edit_resource_instruction"),
-                    "dynamic.resource.editor.enableEditingResource"
-                )
-            }
-        else -> null
+            is ViewEditableDynamicResourceVirtualFile ->
```

Comment:
this seems unrelated ?

Reply:
This is the same code but new condition added where the panel(to create/update) is not displayed if the feature is disabled